mapie/time_series_regression.py

```diff
@@ -0,0 +1,312 @@
+        n_jobs: Optional[int] = None,
+        agg_function: Optional[str] = "mean",
+        verbose: int = 0,
+    ) -> None:
+        super().__init__(estimator, method, cv, n_jobs, agg_function, verbose)
+        self.cv_need_agg_function.append("BlockBootstrap")
+        self.valid_methods_ = ["enbpi"]
+        self.plus_like_method.append("enbpi")
+
+    def _relative_conformity_scores(
+        self,
+        X: NDArray,
+        y: NDArray,
+    ) -> NDArray:
+        """
+        Compute the conformity scores on a data set.
+
+        Parameters
+        ----------
+            X : ArrayLike of shape (n_samples, n_features)
+            Input data.
+
+            y : ArrayLike of shape (n_samples,)
+                Input labels.
+
+        Returns
+        -------
+            The conformity scores corresponding to the input data set.
+        """
+        y_pred, _ = super().predict(X, alpha=0.5, ensemble=True)
+        return np.asarray(y) - np.asarray(y_pred)
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ) -> MapieTimeSeriesRegressor:
+        """
+        Compare to the method ``fit`` of ``MapieRegressor``, the ``fit`` method
+        of ``MapieTimeSeriesRegressor`` computes the ``conformity_scores_``
+        with relative values.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+        """
+        self = super().fit(X=X, y=y, sample_weight=sample_weight)
+        self.conformity_scores_ = self._relative_conformity_scores(X, y)
+        return self
+
+    def partial_fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+    ) -> MapieTimeSeriesRegressor:
+        """
+        Update the ``conformity_scores_`` and ``k_`` attributes when new data
+        with known labels are available.
+        Note: Don't use ``partial_fit`` with samples of the training set.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples_test, n_features)
+            Input data.
+
+        y : ArrayLike of shape (n_samples_test,)
+            Input labels.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+
+        Raises
+        ------
+        ValueError
+            If the lenght of y is greater than the lenght of the training set.
+        """
+        X = cast(NDArray, X)
+        y = cast(NDArray, y)
+        if len(X) > len(self.conformity_scores_):
+            raise ValueError(
+                "You try to update more residuals than there are!"
+            )
+        new_conformity_scores_ = self._relative_conformity_scores(X, y)
+        self.conformity_scores_ = np.roll(
+            self.conformity_scores_, -len(new_conformity_scores_)
+        )
+        self.conformity_scores_[-len(new_conformity_scores_):] = (
+            new_conformity_scores_
+        )
+        return self
+
+    def _beta_optimize(
```

Comment:
Move the "hidden" methods `_beta_optimize` and `_pred_multi` before `fit`.

Reply:
Done